tests/e2e/matmul/CMakeLists.txt

```diff
@@ -2042,7 +2048,8 @@ iree_generated_e2e_runner_test(
 
 ### BAZEL_TO_CMAKE_PRESERVES_ALL_CONTENT_BELOW_THIS_LINE ###
 
-if(IREE_HIP_TEST_TARGET_CHIP MATCHES "^gfx")
+# To distinguish between CDNA and RDNA
+if(IREE_HIP_TEST_TARGET_CHIP MATCHES "^gfx9")
```

Comment:
There is a modifier for gfx9 but not for gfx11? Something is wrong

Reply:
There are rdna tests here:
https://github.com/iree-org/iree/blob/fc3561c1380e85084a8c1584e743264d849aa31d/tests/e2e/matmul/CMakeLists.txt#L1815-L1841

Looks like those should actually use `IREE_HIP_TEST_COMPILER_FLAGS`.

The CDNA tests were failing to compile with the ROCm target chip: https://github.com/iree-org/iree/actions/runs/8988167048/job/24688797939?pr=17298

I guess the RDNA tests compile even with the CDNA target chip?

Reply:
Right, I think we will be fine if we do it this way with bazel, but I would prefer if we did it the same way as CDNA to have consistency.

Reply:
Okay yeah. So

* Remove HIP tests from `BUILD.bazel`
* Add two sections of HIP tests in CMakeLists.txt: `CDNA` and `RDNA`, using the `^gfx9` check and `else()` (or `elseif()`), both using `IREE_HIP_TEST_COMPILER_FLAGS` instead of hardcoded flags like `"--iree-rocm-target-chip=gfx1100"`

Reply:
I made the changes; could you double-check?